idx/widgets/impress-carousel-widget.php

```diff
@@ -160,11 +161,32 @@ public function body( $instance ) {
 
 		$output .= sprintf( '<div class="impress-carousel impress-listing-carousel-%s owl-carousel owl-theme">', $instance['display'] );
 
+		// Used to hold agent data when matching for co-listings.
+		$agent_data;
+
 		foreach ( $properties as $prop ) {
 
-			if ( isset( $instance['agentID'], $prop['userAgentID'] ) && ! empty( $instance['agentID'] ) ) {
-				if ( $instance['agentID'] !== (int) $prop['userAgentID'] ) {
-					continue;
+			if ( ! empty( $instance['agentID'] ) ) {
+				if ( empty( $prop['userAgentID'] ) || (int) $instance['agentID'] !== (int) $prop['userAgentID'] ) {
```

Comment:
It's not too bad here, but a general pattern for reducing on these nested `if`s is to check against the inverse condition then return early, rather than putting in the `else`. Makes it more readable.

For example:
```
if ( empty( $prop['userAgentID'] ) || (int) $instance['agentID'] !== (int) $prop['userAgentID'] ) {
  if ( ! $instance['colistings'] ) {
      continue;
  }
  if ( ! array_key_exists( 'coListingAgentID', $prop ) ) {
    continue;
  }
  // Previously super nested inner if code here.
}
```